Gateway/Validator/InstallmentValidator.php

```diff
@@ -77,7 +78,13 @@ public function validate(array $validationSubject)
         $isValid = true;
         $fails = [];
         $payment = $validationSubject['payment'];
-        $quote = $this->session->getQuote();
+        $quoteId = $payment->getQuoteId();
+        //This validator also runs for other payments that don't necesarily have a quoteId
+        if ($quoteId) {
+            $quote = $this->quoteRepository->get($quoteId);
+        } else {
+            $quote = false;
+        }
```

Comment:
if quote is empty we can ignore it right. Not better to do if(!$quoteId) { return; } ?

Reply:
Goed idee. The validate function returns `$this->createResult($isValid, $fails)`. Discussing with Attila we think that we could make it return that result if there's no quoteId but seems like unnecessarily duplicating code.